### ISSUE TITLE
Information about prediction and training on my dataset

### ISSUE THREAD

**Issue by martirius** [experiencor/keras-yolo2#254]
Hi @experiencor ,
Thanks for your great work.
I have some question about prediction and training.
About training: i've trained on my own dataset(one class, images and bound boxes took from image-net), after 23 fo 50 epoch, the train process stopped saying  this:
Epoch 00023: val_loss did not improve
Epoch 00023: early stopping
/home/mattia/Tensorflow/keras-yolo2/utils.py:198: RuntimeWarning: overflow encountered in exp
  return 1. / (1. + np.exp(-x))
n01443537 0.0000
mAP: 0.0000
and is not encouraging.
However the val_loss and loss were enough low, loss: 0.1420 - val_loss: 0.1616 so i tried the trained weights and surprisingly it works quite well, tried with other classes images and no false positive detected.
I've also modified predict.py for reading frames from a camera and it works but in a strange way: if the camera is really close to the picture(5-6 cm) it detects the object, but if i outdistance the camera from the picture(10 cm and more) it doesn't detect anything. 
Is this the normal behavior of your implementation of YoloV2? Because i didn't found this problem with darknet

**Comment by experiencor** [comment]
There a limitation with the custom sigmoid in utitls. Try to use https://docs.scipy.org/doc/scipy-0.14.0/reference/generated/scipy.special.expit.html instead.

It's not working well if you train it on big objects only. Try to include small objects in the dataset.
